src/SFA.DAS.Commitments.Api.IntegrationTests/DatabaseSetup/Generators/DataLockStatusGenerator.cs

```diff
@@ -0,0 +1,192 @@
+
+namespace SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup.Generators
+{
+    /// <remarks>
+    /// We generate more realistic data than the slowdown test requires, but the idea is that when we add additional integration tests, they can use the generated data.
+    /// If your new test requires more realistic data, then update the generation here to the benefit of all tests.
+    /// </remarks>
+    public class DataLockStatusGenerator : Generator
+    {
+        private static readonly DateTime SystemStartDate = new DateTime(2016, 8, 1);
+        private static readonly int MonthsSinceSystemStartDate = (int)((DateTime.Now - SystemStartDate).TotalDays / (365.25 / 12));
+        private static readonly int SecondsSinceSystemStartDate = (int)(DateTime.Now - SystemStartDate).TotalSeconds;
+        private static readonly string[] ExampleFrameworkCourseCodes = { "583-3-4", "548-3-1", "418-2-1" };
+
+        public async Task<IEnumerable<DbSetupDataLockStatus>> Generate(int numberOfNewApprenticeships, long firstNewApprenticeshipId, long firstNewDataLockEventId)
+        {
+            // generate id groups first with probability distribution, then generate failed from the actual number generated
+            // we lose the absolute ratio of statuses to apprenticeships, but probabilitydistribution will act as the ratio in effect
+            var randomlyOrderedApprenticeshipIdGroups = RandomIdGroups(firstNewApprenticeshipId,
+                numberOfNewApprenticeships, TestDataVolume.DataLockStatusesPerApprenticeshipProbability,
+                (id, count) => GenerateStartDates(count).Select(date => new IdAndDate {Id = id, Date = date}));
+
+            var totalDataLockStatusesToGenerate = randomlyOrderedApprenticeshipIdGroups.Length;
+
+            await TestSetup.LogProgress($"Generating {totalDataLockStatusesToGenerate} DataLockStatuses");
+
+            var errorDataLockStatusesToGenerate = (int)(totalDataLockStatusesToGenerate * TestDataVolume.ErrorDataLockStatusProbability);
+
+            var failedDataLockStatusIds = FailedDataLockStatusIds(firstNewDataLockEventId, totalDataLockStatusesToGenerate, errorDataLockStatusesToGenerate);
+
+            var dataLockEventIds = Enumerable.Range((int)firstNewDataLockEventId, totalDataLockStatusesToGenerate);
+
+            return dataLockEventIds.Zip(randomlyOrderedApprenticeshipIdGroups,
+                (dataLockEventId, apprenticeshipIdAndStartDate) =>
+                    GenerateDbSetupDataLockStatus(dataLockEventId, apprenticeshipIdAndStartDate, failedDataLockStatusIds));
+        }
+
+        private HashSet<long> FailedDataLockStatusIds(long firstNewDataLockEventId, int totalDataLockStatuses,
+            int errorDataLockStatusesToGenerate)
+        {
+            // as we have a potentially large range and require only a small random subset, we do it this way...
+
+            int randomRangeTop = (int) (firstNewDataLockEventId + totalDataLockStatuses);
```

Comment:
This "could" throw an exception if 'firstNewDataLockEventId' is bigger than int.MaxValue

Reply:
See reply to ApprenticeshipUpdateGenerator.cs comment.